src/components/layermanager/LayermanagerDirective.js

```diff
@@ -184,9 +204,132 @@ goog.require('ga_urlutils_service');
           var layersCollection = map.getLayers();
           layersCollection.removeAt(index);
           layersCollection.insertAt(index + delta, layer);
+          scope.tools.index = scope.filteredLayers.indexOf(layer);
           evt.preventDefault();
         };
 
+        scope.onDropComplete = function(currentLayer, evt) {
+          var currentIndex = scope.filteredLayers.indexOf(currentLayer);
+          if (drag.position.direction === drag.under &&
+              currentIndex > drag.position.index) {
+            drag.position.index++;
+          } else if (drag.position.direction === drag.above &&
+              currentIndex < drag.position.index) {
+            drag.position.index--;
+          }
+
+          if (drag.position.index < 0) {
+            drag.position.index = 0;
+          } else if (drag.position.index >= scope.layers.length) {
+            drag.position.index = scope.layers.length - 1;
+          }
+          evt.event.stopPropagation();
+          evt.event.preventDefault();
+          if (currentLayer) {
+            var layersCollection = map.getLayers();
+            var index = scope.layers.indexOf(currentLayer);
+            layersCollection.removeAt(index);
+            var correctedIndex = scope.layers.length - drag.position.index;
+            layersCollection.insertAt(correctedIndex, currentLayer);
```

Comment:
Thanks for the tip. It was part of the problem (with the fact that slipjs fired the reorder event more than once). I am doing more testing, but it looks like the slipjs works really good.